### PR TITLE
Move helper API up as it will be using functionality in EditorFeatures as i refactor more of FAR.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
**Customer scenario**

What does the customer do to get into this situation, and why do we think this
is common enough to address for this release.  (Granted, sometimes this will be
obvious "Open project, VS crashes" but in general, I need to understand how
common a scenario is)

**Bugs this fixes:**

(either VSO or GitHub links)

**Workarounds, if any**

Also, why we think they are insufficient for RC vs. RC2, RC3, or RTW

**Risk**

This is generally a measure our how central the affected code is to adjacent
scenarios and thus how likely your fix is to destabilize a broader area of code

**Performance impact**

(with a brief justification for that assessment (e.g. "Low perf impact because no extra allocations/no complexity changes" vs. "Low")

**Is this a regression from a previous update?**

**Root cause analysis:**

How did we miss it?  What tests are we adding to guard against it in the future?

**How was the bug found?**

(E.g. customer reported it vs. ad hoc testing)

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit32_prtest  please

File: src/VisualStudio/Core/Test/FindResults/FindResultsTests.vb [hunk 1]
```diff
@@ -13,6 +13,7 @@ Imports Microsoft.VisualStudio.LanguageServices.Implementation.Library.FindResul
 Imports Microsoft.VisualStudio.LanguageServices.UnitTests.ObjectBrowser.Mocks
 Imports Roslyn.Test.Utilities
 Imports Roslyn.Utilities
+Imports Microsoft.CodeAnalysis.Editor.FindUsages
```
Comment (sharwell) [inline src/VisualStudio/Core/Test/FindResults/FindResultsTests.vb]: :bulb: Ordering

**Comment by sharwell** [comment]
Blocked on #18840.

File: src/EditorFeatures/Core/FindUsages/IDefinitionsAndReferencesFactory.cs [hunk 2]
```diff
@@ -8,12 +8,13 @@
...
 using Microsoft.CodeAnalysis.Completion;
 using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.FindUsages;
 using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Host.Mef;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.FindUsages
+namespace Microsoft.CodeAnalysis.Editor.FindUsages
```
Comment (dpoeschl) [inline src/EditorFeatures/Core/FindUsages/IDefinitionsAndReferencesFactory.cs]: This is the only actual code change for this PR, right?

**Comment by CyrusNajmabadi** [comment]
Blocking issue has been merged in.

File: src/EditorFeatures/Core/FindUsages/IDefinitionsAndReferencesFactory.cs (hunk 2 above)
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/FindUsages/IDefinitionsAndReferencesFactory.cs]: Yes...?

**Comment by CyrusNajmabadi** [comment]
Tagging @MattGertz Continued refactoring to enable FAR-OOP.  This one specifically enables us to compute things like how the final text should be classified so we don't have go back to expensive symbolic computation on the VS side to figure that out.

**Pull request merged**